File: app/src/main/java/com/denchic45/kts/uivalidator/validatorlistener/FieldMessageLiveDataValidationListener.java
package com.denchic45.kts.uivalidator.validatorlistener;

import androidx.lifecycle.MutableLiveData;

import com.denchic45.kts.Pair;
import com.denchic45.kts.uivalidator.Rule;

public class FieldMessageLiveDataValidationListener extends LiveDataValidationListener<Pair<Integer, String>> {

    private final int id;

    public FieldMessageLiveDataValidationListener(int id, MutableLiveData<Pair<Integer, String>> mutableLiveData) {
        super(mutableLiveData);
        this.id = id;
    }

    @Override
    void onSuccess() {
        mutableLiveData.setValue(new Pair<>(id, null));
    }

    @Override
    void onError(Rule rule) {
        mutableLiveData.setValue(new Pair<>(id, rule.getErrorMessage()));
    }
}

File: app/src/main/java/com/denchic45/kts/uivalidator/validatorlistener/FieldMessageLiveDataValidationListener.kt
package com.denchic45.kts.uivalidator.validatorlistener

import com.denchic45.kts.SingleLiveData
import com.denchic45.kts.uivalidator.Rule

class FieldMessageLiveDataValidationListener(
    private val id: Int,
    mutableLiveData: SingleLiveData<Pair<Int, String?>>
) : LiveDataValidationListener<Pair<Int, String?>>(mutableLiveData) {
    override fun onSuccess() {
        mutableLiveData.value = Pair<Int, String?>(id, null)
    }

    override fun onError(rule: Rule) {
        mutableLiveData.value = Pair(id, rule.errorMessage)
    }
}